arangod/ClusterEngine/ClusterCollection.cpp

```diff
@@ -382,16 +383,14 @@ bool ClusterCollection::dropIndex(TRI_idx_iid_t iid) {
     return true;
   }
 
-  size_t i = 0;
   WRITE_LOCKER(guard, _indexesLock);
-  for (std::shared_ptr<Index> index : _indexes) {
-    if (iid == index->id()) {
-      _indexes.erase(_indexes.begin() + i);
+  for (auto it  : _indexes) {
+    if (iid == it->id()) {
+      _indexes.erase(it);
```

Comment:
Doesn't this invalidate the iterator `it` that is used in the `for` loop around?
Is that safe to do?

Reply:
Yes, it invalidates. But it doesn`t matter as loop terminates after that.

Reply:
True